autoload/ale/fix.vim

```diff
@@ -457,13 +457,17 @@ function! ale#fix#Fix(...) abort
     call ale#fix#RemoveManagedFiles(l:buffer)
     call ale#fix#InitBufferData(l:buffer, l:fixing_flag)
 
+    silent doautocmd <nomodeline> User ALEFixPre
+
     call s:RunFixer({
     \   'buffer': l:buffer,
     \   'input': g:ale_fix_buffer_data[l:buffer].lines_before,
     \   'callback_index': 0,
     \   'callback_list': l:callback_list,
     \})
 
+    silent doautocmd <nomodeline> User ALEFixPost
```

Comment:
This line should be moved to `ApplyFixes` or at the end of `ApplyQueuedFixes`, but before linting is queued again. The former would execute when ALE is ready to replace a buffer's lines. The latter would execute after ALE has actually replaced the lines. Vim buffers can't be edited unless you're looking at them, so the the changes aren't applied until the buffer is opened again, which isn't often a problem as things tend to get fixed while you're still looking at the buffer. I'd recommend the latter option.

Sorry about the delay. I was on holiday.

Reply:
Thanks for the feedback :+1:

I moved the line and tested again. Seems to work well. Also rebased the branch on master as it's been a while.